### PR TITLE
Adding comments to Plan

### LINKED ISSUE #82: Commenting feature

### PR DESCRIPTION
**Pull request by Vielheim**

### CODE REVIEW

**Comment by Vielheim** [comment]
Add new collection: comments
- Comment model for Domain and Data representation
- New services for Comments

Add CommentsViewModel
- In charge of CRUD of comments
- Previously we separated the Add and Fetch/Edit/Delete into two VMs. But for comments because they are all embedded inside one View, I left the CRUD inside one VM. 


 Added Views:
- CommentsView: Parent view for each comment section.
- CommentListView: Renders a list of comments, sorted by date
- Individual CommentView. This view takes in the CommentsVM, a Comment and a User to render. I wanted to put the (Comment, User) inside a smaller CommentVM, but I am not able to get a "real-time" update whenever we edit or delete the comment. When I used the CommentsVM, I could decide when to re-render the entire Comments Section for a more real-time sort of update.
- EditCommentView - Create a `.sheet` like how we did for the other Models

File: TourMate/TourMate/Backend/Persistence/Config/FirebaseConfig.swift [hunk 1]
```diff
@@ -12,6 +12,7 @@ final class FirebaseConfig {
     static let tripCollectionId = "trips"
     static let planCollectionId = "plans"
     static let planDetailsCollectionId = "plandetails"
```
Comment (keanecjy) [inline TourMate/TourMate/Backend/Persistence/Config/FirebaseConfig.swift:14]: can remove the plandetails above

**Review by keanecjy — COMMENTED** [review]
1. Can bold the like button when the user has liked it so that the user can tell he has liked it.
2. The box gets bigger when the like button shows

**Comment by Vielheim** [comment]
> * Can bold the like button when the user has liked it so that the user can tell he has liked it.
> * The box gets bigger when the like button shows

Fixed the box padding issue.
The bolding for text is not very obvious. But when you like the comment, the thumbsup icon is blue. If you haven't and someone else liked it, the icon is white.

**Review by keanecjy — APPROVED** [review]
LGTM

**Pull request merged**